app/services/streaming/streaming.ts

```diff
@@ -267,34 +227,79 @@ export class StreamingService extends StatefulService<IStreamingServiceState> im
     return `${hours}:${minutes}:${seconds}`;
   }
 
-  private emitStateIfChanged() {
-    if (isEqual(this.state, this.emittedState)) return;
-    const stateToEmit = { ...this.state };
-    this.streamingStateChange.next(this.state);
-    this.emittedState = stateToEmit;
-  }
-
-  get delayEnabled() {
-    return this.settingsService.state.Advanced.DelayEnable;
-  }
-
-  get delaySeconds() {
-    return this.settingsService.state.Advanced.DelaySec;
-  }
+  private handleOBSOutputSignal(info: IOBSOutputSignalInfo) {
+    console.debug('OBS Output sigal: ', info);
+    if (info.type === EOBSOutputType.Streaming) {
+      const time = new Date().toISOString();
+
+      if (info.signal === EOBSOutputSignal.Start) {
+        this.SET_STREAMING_STATUS(EStreamingState.Live, time);
+        this.streamingStatusChange.next(EStreamingState.Live);
+      } else if (info.signal === EOBSOutputSignal.Starting) {
+        this.SET_STREAMING_STATUS(EStreamingState.Starting, time);
+        this.streamingStatusChange.next(EStreamingState.Starting);
+      } else if (info.signal === EOBSOutputSignal.Stop) {
+        this.SET_STREAMING_STATUS(EStreamingState.Offline, time);
+        this.streamingStatusChange.next(EStreamingState.Offline);
+      } else if (info.signal === EOBSOutputSignal.Stopping) {
+        this.SET_STREAMING_STATUS(EStreamingState.Ending, time);
+        this.streamingStatusChange.next(EStreamingState.Ending);
+      }
+    } else if (info.type === EOBSOutputType.Recording) {
+      const time = new Date().toISOString();
+
+      if (info.signal === EOBSOutputSignal.Start) {
+        this.SET_RECORDING_STATUS(ERecordingState.Recording, time);
+      } else if (info.signal === EOBSOutputSignal.Starting) {
+        this.SET_RECORDING_STATUS(ERecordingState.Starting, time);
+      } else if (info.signal === EOBSOutputSignal.Stop) {
+        this.SET_RECORDING_STATUS(ERecordingState.Offline, time);
+      } else if (info.signal === EOBSOutputSignal.Stopping) {
+        this.SET_RECORDING_STATUS(ERecordingState.Stopping, time);
```

Comment:
Don't we expose recording events for Streamdeck?

Reply:
We did, but they don't appear to be using it.  It might be worth adding back in though just to match the streaming output.